internal/execute/tableutil/builder_cache.go

```diff
@@ -0,0 +1,169 @@
+package tableutil
+
+import (
+	"reflect"
+
+	"github.com/influxdata/flux"
+	"github.com/influxdata/flux/codes"
+	"github.com/influxdata/flux/execute"
+	"github.com/influxdata/flux/internal/errors"
+	"github.com/influxdata/flux/plan"
+)
+
+// Constructor is the minimum interface for constructing a Table.
+type Constructor interface {
+	// Table will construct a Table from the existing contents.
+	// Invoking this method should reset the builder and all allocated
+	// memory will be owned by the returned flux.Table.
+	Table() (flux.Table, error)
+
+	// Release will release the buffered contents from the builder.
+	// This method is unnecessary if Table is called.
+	Release()
+}
+
+type BuilderContext struct {
+	Builder Constructor
+	Trigger execute.Trigger
+}
+
+// BuilderCache is an alternative table builder cache.
+// It allows a pluggable implementation of the builder.
+// The only requirement for the builder is that it must
+// implement the above Constructor interface.
+type BuilderCache struct {
+	New func(key flux.GroupKey) Constructor
+
+	tables      *execute.GroupLookup
+	triggerSpec plan.TriggerSpec
+}
```

Comment:
What does "pluggable" mean in this context?

Reply:
It allows us to return any arbitrary interface as long as it can produce a table and can have its contents released. The transformation itself would know which version it creates because it sets the `New` method, but that would allow this builder cache to be used with any builder implementation so it is easier for us to experiment with different implementations.

Inside of this PR, I had created both one for constructing a table directly with arrow builders and one that created a table out of multiple compatible buffers. Both are able to use this underlying implementation for some of the heavy lifting to reduce common code.